File: frontend/src/services/supplierService.js
import api from './api';

export const supplierService = {
  /**
   * Get paginated list of suppliers
   */
  async getSuppliers(params = {}) {
    const { page = 1, pageSize = 50 } = params;
    
    const queryParams = new URLSearchParams({
      page: page.toString(),
      page_size: pageSize.toString(),
    });

    const response = await api.get(`/suppliers?${queryParams.toString()}`);
    return response.data;
  },

  /**
   * Get a single supplier by ID
   */
  async getSupplier(id) {
    const response = await api.get(`/suppliers/${id}`);
    return response.data;
  },

  /**
   * Create a new supplier
   */
  async createSupplier(data) {
    const response = await api.post('/suppliers', data);
    return response.data;
  },

  /**
   * Update an existing supplier
   */
  async updateSupplier(id, data) {
    const response = await api.put(`/suppliers/${id}`, data);
    return response.data;
  },

  /**
   * Delete a supplier
   */
  async deleteSupplier(id) {
    await api.delete(`/suppliers/${id}`);
  },
};

export default supplierService;

File: frontend1/lib/services/supplier_service.dart
import 'package:frontend1/services/api_service.dart';
import 'package:frontend1/models/supplier.dart';

class SupplierService {
  final ApiService _apiService = ApiService();

  Future<SupplierResponse> getSuppliers({int page = 1, int pageSize = 10, String? search}) async {
    final Map<String, dynamic> queryParams = {
      'page': page,
      'size': pageSize,
    };
    if (search != null && search.isNotEmpty) {
      queryParams['search'] = search;
    }

    final response = await _apiService.get('/suppliers', queryParameters: queryParams);
    return SupplierResponse.fromJson(response.data);
  }

  Future<Supplier> createSupplier(Map<String, dynamic> data) async {
    final response = await _apiService.post('/suppliers', data: data);
    return Supplier.fromJson(response.data);
  }

  Future<Supplier> updateSupplier(int id, Map<String, dynamic> data) async {
    final response = await _apiService.put('/suppliers/$id', data: data);
    return Supplier.fromJson(response.data);
  }

  Future<void> deleteSupplier(int id) async {
    await _apiService.delete('/suppliers/$id');
  }
}
